src/test_application/VehicleBuilder.h

```diff
@@ -52,115 +64,62 @@ class VehicleBuilder
+    template <std::size_t N>
+    [[nodiscard]] std::array<PartId, N> create_parts();
 
-    template<typename WIRETYPE_T>
-    nodeindex_t<WIRETYPE_T> wire_node_add(osp::ConfigNode_t config)
+    struct SetPrefab
     {
-        return wire_node_add(osp::wiretype_id<WIRETYPE_T>(), config);
-    }
+        PartId m_part;
+        std::string_view m_prefabName;
+    };
 
-    linkindex_t<void> wire_connect(
-            osp::wire_id_t id, nodeindex_t<void> node, osp::NodeMap_t config,
-            part_t part, mach_t mach, portindex_t<void> port);
+    void set_prefabs(std::initializer_list<SetPrefab> setPrefab);
 
-    template<typename WIRETYPE_T>
-    linkindex_t<WIRETYPE_T> wire_connect(
-            nodeindex_t<WIRETYPE_T> node, osp::NodeMap_t config,
-            part_t part, mach_t mach, portindex_t<WIRETYPE_T> port)
+    struct SetTransform
     {
-        return linkindex_t<WIRETYPE_T>(wire_connect(osp::wiretype_id<WIRETYPE_T>(), config, part, mach, port));
-    }
+        PartId m_part;
+        osp::Matrix4 const& m_transform;
+    };
 
-    nodeindex_t<void> wire_connect_signal(osp::wire_id_t id,
-            part_t partOut, mach_t machOut, portindex_t<void> portOut,
-            part_t partIn, mach_t machIn, portindex_t<void> portIn);
+    void set_transform(std::initializer_list<SetTransform> setTransform);
 
-    template<typename WIRETYPE_T>
-    nodeindex_t<WIRETYPE_T> wire_connect_signal(
-            part_t partOut, mach_t machOut, portindex_t<WIRETYPE_T> portOut,
-            part_t partIn, mach_t machIn, portindex_t<WIRETYPE_T> portIn)
-    {
-        return nodeindex_t<WIRETYPE_T>(wire_connect_signal(osp::wiretype_id<WIRETYPE_T>(), partOut, machOut, portindex_t<void>(portOut), partIn, machIn, portindex_t<void>(portIn)));
-    }
+    osp::Matrix4 align_attach(PartId partA, std::string_view attachA,
+                              PartId partB, std::string_view attachB);
 
-    osp::BlueprintVehicle&& export_move() { return std::move(m_vehicle); };
-    osp::BlueprintVehicle export_copy() { return m_vehicle; };
+    VehicleData const& data() const noexcept { return m_data; }
 
 private:
 
-    // Map used to obtain a Machine's Panel
-    // panelIndex = m_panelMap[{id, part, machine}]
-    // panel = m_vehicle.m_wirePanels[panelIndex]
-    std::map<std::tuple<osp::wire_id_t, part_t, mach_t>,
-             uint32_t> m_panelIndexMap;
+    void index_prefabs();
+
+    osp::Resources *m_pResources;
 
-    // Map used to obtain Nodes from part
-    // nodeIndex = m_nodeIndexMap[{id, part, machine, port}]
-    std::map<std::tuple<osp::wire_id_t, part_t, mach_t, portindex_t<void>>,
-             nodeindex_t<void>> m_nodeIndexMap;
+    entt::dense_hash_map< std::string_view, osp::PrefabPair > m_prefabs;
 
-    osp::BlueprintVehicle m_vehicle;
+    VehicleData m_data;
 
+    // put more attachment data here
 };
 
 
+template <std::size_t N>
+[[nodiscard]] std::array<PartId, N> VehicleBuilder::create_parts()
```

Comment:
you only need to annotate the declaration with [[nodiscard]].

No need to put it on the definition.,